pkg/policy/validate.go

```diff
@@ -187,6 +209,22 @@ func Validate(policy *kyverno.ClusterPolicy, client *dclient.Client, mock bool,
 // returns true if it is an empty set
 func doMatchAndExcludeConflict(rule kyverno.Rule) bool {
 
+	if len(rule.MatchResources.All) > 0 || len(rule.ExcludeResources.All) > 0 {
+		return false
+	}
+
+	// if both have any then no resource should be common
+	if len(rule.MatchResources.Any) > 0 && len(rule.ExcludeResources.Any) > 0 {
+		for _, rmr := range rule.MatchResources.Any {
+			for _, rer := range rule.ExcludeResources.Any {
+				if reflect.DeepEqual(rmr, rer) {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
```

Comment:
Covered by [this](https://github.com/kyverno/kyverno/pull/2130/files#diff-c980abe34c6a2660ca3281d07c6f8721eebc10133260700e13b208dbe4259986R208-R213) since we use **OR**.

Reply:
As discussed, let's remove this check and log a separate issue.
```
	if len(rule.ExcludeResources.All) > 0 || len(rule.MatchResources.All) > 0 {
		return false
	}
```

Reply:
If we remove this then the policies with "all" won't get applied since this function needs to return `false` in order for the policy to get applied.

Reply:
I thought by default we return false. Then let's add a comment pointing to the issue https://github.com/kyverno/kyverno/issues/2203.